Sources/Signal.swift

```diff
@@ -571,6 +571,31 @@ extension SignalProtocol {
 			}
 		}
 	}
+	
+	/// Applies `transform` to values from `signal` and forwards values with non `nil` results unwrapped.
+	/// - parameters:
+	///   - transform: A closure that accepts a value from the `value` event and
+	///                returns a new optional value.
+	///
+	/// - returns: A signal that will send new values, that are non `nil` after the transformation.
+	public func filterMap<U>(_ transform: @escaping (Value) -> U?) -> Signal<U, Error> {
+		return Signal { observer in
```

Comment:
It's been a couple years since I've tested it, but we noticed similar lag on larger iOS scroll views with deep signal chains. Hence why I went with the direct implementation here.